File: house-robber/house-robber.c
int max(int x, int y){
    if (x>y)
        return x;
    else
        return y;
}

int rob(int* nums, int numsSize){
    
    if(nums==NULL || numsSize<1) return 0;
    
    // int* dp=(int*)malloc(numsSize*2*sizeof(int));
    
    
    
    int *dp[numsSize];
    for (int i=0; i<numsSize; i++)
         dp[i] = (int *)malloc(2 * sizeof(int));
    
    dp[0][0]=0;
    dp[0][1]=nums[0];
    
    for(int i=1;i<numsSize;i++){
        dp[i][0]=max(dp[i-1][0],dp[i-1][1]);
        dp[i][1]=dp[i-1][0] + nums[i];
    }
    
    return max(dp[numsSize-1][0],dp[numsSize-1][1]);
    
    return 0;

}

File: house-robber/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        if nums is None : 
            return 0;
        
        if len(nums) is 1:
            return nums[0];
        
        n=len(nums);
        
        # rows,cols=(n,2)
        dp=[[0 for i in range(2)] for j in range(n)]
        
        dp[0][1]=nums[0]
        
        for i in range(1,n):
            dp[i][0]= max(dp[i-1][0],dp[i-1][1])
            dp[i][1]= dp[i-1][0]+nums[i]
            
        return max(dp[n-1][0],dp[n-1][1])
        
        
        
        
        